tests/contrib/io/matplotlib/test_matplotlib_s3_writer.py

```diff
@@ -0,0 +1,238 @@
+
+        mock_list_plot[index].savefig(str(actual_filepath))
+
+        _key_path = KEY_PATH + "/" + str(index) + ".png"
+
+        mocked_s3_bucket.download_file(BUCKET_NAME, _key_path, str(expected_path))
+
+        assert actual_filepath.read_bytes() == expected_path.read_bytes()
+
+
+def test_dict_save(tmp_path, mock_dict_plot, plot_writer, mocked_s3_bucket):
+    """Test saving dictionary of plots to S3."""
+
+    plot_writer.save(mock_dict_plot)
+
+    for colour in COLOUR_LIST:
+
+        expected_path = tmp_path / "downloaded_image.png"
+        actual_filepath = tmp_path / "locally_saved.png"
+
+        mock_dict_plot[colour].savefig(str(actual_filepath))
+
+        _key_path = KEY_PATH + "/" + colour
+
+        mocked_s3_bucket.download_file(BUCKET_NAME, _key_path, str(expected_path))
+
+        assert actual_filepath.read_bytes() == expected_path.read_bytes()
+
+
+def test_bad_credentials(mock_dict_plot):
+    """Test writing with bad credentials"""
+    bad_writer = MatplotlibS3Writer(
+        bucket_name=BUCKET_NAME,
+        filepath=KEY_PATH,
+        credentials={
+            "aws_access_key_id": "not_for_testing",
+            "aws_secret_access_key": "definitely_not_for_testing",
+        },
+    )
+    pattern = "InvalidAccessKeyId"
+
+    with pytest.raises(DataSetError, match=pattern):
+        bad_writer.save(mock_dict_plot)
+
+
+def test_credentials(tmp_path, mock_single_plot, mocked_s3_bucket):
+    """Test entering credentials"""
+    normal_writer = MatplotlibS3Writer(
+        bucket_name=BUCKET_NAME, filepath=KEY_PATH, credentials=AWS_CREDENTIALS
+    )
+
+    normal_writer.save(mock_single_plot)
+
+    expected_path = tmp_path / "downloaded_image.png"
+    actual_filepath = tmp_path / "locally_saved.png"
+
+    plt.savefig(str(actual_filepath))
+
+    mocked_s3_bucket.download_file(BUCKET_NAME, KEY_PATH, str(expected_path))
+
+    assert actual_filepath.read_bytes() == expected_path.read_bytes()
+
+
+def test_s3_encryption(tmp_path, mock_single_plot, mocked_encrypted_s3_bucket):
+    """Test writing to encrypted bucket"""
+    normal_encryped_writer = MatplotlibS3Writer(
+        bucket_name=BUCKET_NAME,
+        s3fs_args={"s3_additional_kwargs": {"ServerSideEncryption": "AES256"}},
+        filepath=KEY_PATH,
+        credentials=AWS_CREDENTIALS,
+    )
+
+    normal_encryped_writer.save(mock_single_plot)
+
+    expected_path = tmp_path / "downloaded_image.png"
+    actual_filepath = tmp_path / "locally_saved.png"
+
+    mock_single_plot.savefig(str(actual_filepath))
+
+    mocked_encrypted_s3_bucket.download_file(BUCKET_NAME, KEY_PATH, str(expected_path))
+
+    assert actual_filepath.read_bytes() == expected_path.read_bytes()
+
+
+def test_load_fail(plot_writer):
+    pattern = r"Loading not supported for `MatplotlibS3Writer`"
+    with pytest.raises(DataSetError, match=pattern):
+        plot_writer.load()
+
+
+def test_exists_single(mock_single_plot, plot_writer):
+    plot_writer.save(mock_single_plot)
```

Comment:
You can do `assert not plot_writer.exists()` before this line. As for the next test.

Reply:
Done!